Sources/TSCBasic/FileSystem.swift

```diff
@@ -441,7 +441,14 @@ private class LocalFileSystem: FileSystem {
 
     func removeFileTree(_ path: AbsolutePath) throws {
         if self.exists(path, followSymlink: false) {
-            try FileManager.default.removeItem(atPath: path.pathString)
+            do {
+                try FileManager.default.removeItem(atPath: path.pathString)
+            } catch let error as NSError {
+                // If we failed because the directory doesn't actually exist anymore, ignore the error.
+                if !(error.domain == NSPOSIXErrorDomain && error.code == 2 /* ENOENT */) {
```

Comment:
do we know if this is correct across the different platforms? cc @compnerd

Reply:
Looks like there's actually a constant for this, I'll update. https://github.com/apple/swift-corelibs-foundation/blob/e9d8347e64ece3de453d617887af7229f73358d8/Sources/Foundation/FoundationErrors.swift#L15

Reply:
Actually there is also `NSFileReadNoSuchFileError`, it's a bit unclear to me which one would happen here and if it is consistent per platform. Maybe extracting the classification of the error and writing a unit test for that could make sense?

Reply:
Good thing I added the test, I was mistakenly looking at the underlying error when figuring out what to look for. So we actually get an error with `NSCocoaErrorDomain` with code `NSFileNoSuchFileError` but it has an underlying error of `NSPOSIXErrorDomain` with code 2 (which is `ENOENT`). The Cocoa domain one seems to be consistent.

There is this quirk on Windows: https://github.com/apple/swift-corelibs-foundation/blob/e9d8347e64ece3de453d617887af7229f73358d8/Sources/Foundation/FoundationErrors.swift#L229-L233, so technically we are changing the behavior for `removeFileTree("")` on that platform, but I think that's fine?

Reply:
The error domains are different IIRC.  Sounds like the condition is being removed though

Reply:
> so technically we are changing the behavior for removeFileTree("") on that platform, but I think that's fine?

in TSC scope? I think that is fine

Reply:
@compnerd Oh I see what you are saying, there's a special Windows domain [here](https://github.com/apple/swift-corelibs-foundation/blob/e9d8347e64ece3de453d617887af7229f73358d8/Sources/Foundation/FoundationErrors.swift#L212). I guess we could not check for the domain at all and just check for `NSFileNoSuchFileError` as the code?

Reply:
Ah wait, I was misreading it, the special domain is only for the [underlying error](https://github.com/apple/swift-corelibs-foundation/blob/e9d8347e64ece3de453d617887af7229f73358d8/Sources/Foundation/FoundationErrors.swift#L274), the outer error uses the cocoa domain. So this should be fine for Windows, too.